spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -128,17 +107,16 @@ public MessageProperties toMessageProperties(AMQP.BasicProperties source, Envelo
 
 	private RabbitExtendedBindingProperties extendedBindingProperties = new RabbitExtendedBindingProperties();
 
-	public RabbitMessageChannelBinder(ConnectionFactory connectionFactory, RabbitProperties rabbitProperties) {
-		super(true, new String[0]);
+	RabbitExchangeQueueProvisioner provisioningProvider;
```

Comment:
When the above mentioned issue is addressed - https://github.com/spring-cloud/spring-cloud-stream-binder-rabbit/issues/42 - I think we don't need to do this `autoDeclareContext` cleanup. This is the only reason why we need to provide a reference to `provisioningProvider` in the rabbit binder and rabbit is the only binder implementation that requires this reference today. Due to this, I think we can leave it as it is and when this issue is addressed we can come back and remove the `provisioningProvider` completely from the binder. Thoughts?